code/game/objects/items/devices/radio/radio.dm

```diff
@@ -225,6 +225,7 @@
 		spans = list(M.speech_span)
 	if(!language)
 		language = M.get_selected_language()
+	SEND_SIGNAL(src, COMSIG_RADIO_MESSAGE, M, message)
```

Comment:
In your signal handler, you have `mob/living/user, message, channel` as the parameters, but here you do not send the channel so it will not work.